### PR TITLE
taup: Add tutorial for taup_setsac

### PR DESCRIPTION
**Pull request by seisman**
Related to #1

Modified from https://blog.seisman.info/taup-mark-traveltime/.

### CODE REVIEW

File: source/taup/taup_setsac.rst [hunk 1]
```diff
@@ -0,0 +1,31 @@
+taup_setsac
+===========
+
+:doc:`taup_time` 命令可以计算各震相的理论走时，
+:doc:`taup_setsac` 命令则用于计算理论走时并将走时信息写入到 SAC 文件头段中。
+
+使用 ``taup_setsac --help`` 可查询该命令的完整用法和选项。
+
+其用法很简单，示例如下::
```
Comment (core-man) [inline source/taup/taup_setsac.rst:9]:
```suggestion
其用法相对简单，示例如下::
```

File: source/taup/taup_setsac.rst [hunk 2]
```diff
@@ -0,0 +1,31 @@
+taup_setsac
+===========
+
+:doc:`taup_time` 命令可以计算各震相的理论走时，
+:doc:`taup_setsac` 命令则用于计算理论走时并将走时信息写入到 SAC 文件头段中。
+
+使用 ``taup_setsac --help`` 可查询该命令的完整用法和选项。
+
+其用法相对简单，示例如下::
```
Comment (core-man) [inline source/taup/taup_setsac.rst:9]:
```suggestion
使用 ``taup_setsac --help`` 可查询该命令的完整用法和选项，
其用法相对简单，示例如下::
```

File: source/taup/taup_setsac.rst [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+   $ taup_setsac -mod prem -ph P-0,S-1,PcP-8,ScS-9 -evdpkm *.SAC
```
Comment (core-man) [inline source/taup/taup_setsac.rst:11]:
```suggestion
   $ taup_setsac -mod prem -evdpkm -ph P-0,S-1,PcP-8,ScS-9 *.SAC
```

File: source/taup/taup_setsac.rst [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+-  ``-mod`` 指定使用 PREM 模型
+-  ``-ph P-0,S-1,PcP-8,ScS-9`` 将 P 波到时标记到 SAC 头段 ``T0`` 中，
+   S 到时标记到 ``T1`` 中，PcP 标记到 ``T8``，ScS 标记到 ``T9``。
+-  ``-evdpkm``\ ：由于历史原因，该命令默认 SAC 文件中的震源深度单位为米，
+   而现在的 SAC 文件中的震源深度单位为千米。该选项设置深度的单位为 km
```
Comment (core-man) [inline source/taup/taup_setsac.rst:17]:
```suggestion
-  ``-mod``\ ：指定使用 PREM 模型，默认使用 iasp91 模型。
-  ``-ph P-0,S-1,PcP-8,ScS-9``\ ：将 P 波到时标记到 SAC 头段 ``T0`` 中，
   S 波到时标记到 ``T1`` 中，PcP 到时标记到 ``T8``，ScS 标记到到时 ``T9``。
-  ``-evdpkm``\ ：由于历史原因，该命令默认 SAC 文件中的震源深度单位为米，
   而现在的 SAC 文件中的震源深度单位为千米。该选项设置深度的单位为 km。
```

File: source/taup/taup_setsac.rst [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+其中 ``KTn`` 中保存震相名，``USERn`` 中保存震相的射线参数（单位为 s/radian）。
+
+.. note::
+
+   若执行命令时经常会忘记指定 ``-evdpkm`` 选项，此时该命令会给出警告。
+   需要注意，一定不要按 :kbd:`Ctrl+C` 终止执行命令。
+
+   因为 ``taup_setsac`` 需要将 SAC 文件读入内存，进行修改，再将内存中的数据
+   写回磁盘覆盖原文件。使用 :kbd:`Ctrl+C` 终止命令会导致文件写回磁盘失败，
+   进而导致部分 SAC 文件损坏。
```
Comment (core-man) [inline source/taup/taup_setsac.rst:29]:
```suggestion
   若执行命令时经常会忘记指定 ``-evdpkm`` 选项，此时若震源深度小于 1 km （1000 m），
   该命令会给出警告。
   
   需要注意，一定不要按 :kbd:`Ctrl+C` 终止执行命令。因为 ``taup_setsac`` 
   需要将 SAC 文件读入内存，进行修改，再将内存中的数据回磁盘覆盖原文件。
   使用 :kbd:`Ctrl+C` 终止命令会导致文件写回磁盘失败，进而导致部分 SAC 文件损坏。
```

File: source/taup/taup_setsac.rst [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+   正确的做法是不理会这些警告，待程序执行完毕之后再加上``-evdpkm`` 选项重新执行。
```
Comment (core-man) [inline source/taup/taup_setsac.rst:31]:
```suggestion
   正确的做法是不理会这些警告，待程序执行完毕之后再加上 ``-evdpkm`` 选项重新执行。
```

File: source/taup/taup_setsac.rst [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+-  ``-mod`` 指定使用 PREM 模型
+-  ``-ph P-0,S-1,PcP-8,ScS-9`` 将 P 波到时标记到 SAC 头段 ``T0`` 中，
+   S 到时标记到 ``T1`` 中，PcP 标记到 ``T8``，ScS 标记到 ``T9``。
+-  ``-evdpkm``\ ：由于历史原因，该命令默认 SAC 文件中的震源深度单位为米，
+   而现在的 SAC 文件中的震源深度单位为千米。该选项设置深度的单位为 km
+
+该命令在将震相走时标记到头段变量 ``Tn`` 的同时，还会修改对应的 ``KTn`` 和 ``USERn``，
+其中 ``KTn`` 中保存震相名，``USERn`` 中保存震相的射线参数（单位为 s/radian）。
+
```
Comment (core-man) [inline source/taup/taup_setsac.rst:21]:
```suggestion

使用该命令时，SAC 头段变量 ``EVDP`` 和 ``O` 必须已赋值。
若头段变量 ``GCARC`` 或 ``DIST`` 未赋值，但震源和台站位置的头段变量
（``EVLA``、``EVLO``、``STLA``、``STLO``）已赋值，该命令可以自动计算震中距，
与 :doc:`taup_time` 命令类似。
```

Comment (seisman) [reply source/taup/taup_setsac.rst:20]: > 函数位于 SAC 源码 ``src/ucf/distaz.c`` 中。

I don't think TauP has this code.

Comment (core-man) [reply source/taup/taup_setsac.rst:20]: This is SAC's code. I mean when we write evla evlo stla stlo to SAC header, SAC will call this fucntion to update gcarc and dist.

Comment (core-man) [reply source/taup/taup_setsac.rst:20]: Do we have to revise this comment?

File: source/taup/taup_setsac.rst [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+使用 ``taup_setsac --help`` 可查询该命令的完整用法和选项，
+其用法相对简单，示例如下::
+
+   $ taup_setsac -mod prem -evdpkm -ph P-0,S-1,PcP-8,ScS-9 *.SAC
+
+-  ``-mod``\ ：指定使用 PREM 模型，默认使用 iasp91 模型。
+-  ``-ph P-0,S-1,PcP-8,ScS-9``\ ：将 P 波到时标记到 SAC 头段 ``T0`` 中，
+   S 波到时标记到 ``T1`` 中，PcP 到时标记到 ``T8``，ScS 到时标记到 ``T9``。
+-  ``-evdpkm``\ ：由于历史原因，该命令默认 SAC 文件中的震源深度单位为米，
+   而现在的 SAC 文件中的震源深度单位为千米。该选项设置深度的单位为 km。
```
Comment (core-man) [inline source/taup/taup_setsac.rst:16]:
```suggestion
-  ``-evdpkm``\ ：由于历史原因，该命令默认 SAC 文件中的震源深度单位为米，
   而现在的 SAC 文件中的震源深度单位为千米。该选项设置深度的单位为 km。
-  ``-ph P-0,S-1,PcP-8,ScS-9``\ ：将 P 波到时标记到 SAC 头段 ``T0`` 中，
   S 波到时标记到 ``T1`` 中，PcP 到时标记到 ``T8``，ScS 到时标记到 ``T9``。
```

File: source/taup/taup_setsac.rst [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+（``EVLA``、``EVLO``、``STLA``、``STLO``）已赋值，该命令可以自动计算震中距。
+需要注意，这时震中距的计算有一定的误差，与 :doc:`taup_time` 命令类似。
+不过 SAC 头段变量 ``LCALDA`` 一般设置为 ``TURE``，
+当事件和台站的坐标被写入或被修改时，头段变量 ``GCARC``、``DIST`` 会自动计算，
+而 SAC 计算震中距时考虑了地球的扁率（赤道半径 6378.160 km，扁率为 0.00335293）,
+函数位于 SAC 源码 ``src/ucf/distaz.c`` 中。
+
+.. note::
+
+   若执行命令时经常会忘记指定 ``-evdpkm`` 选项，此时若震源深度小于 1 km （1000 m），
```
Comment (core-man) [inline source/taup/taup_setsac.rst:33]:
```suggestion
   若执行命令时忘记了指定 ``-evdpkm`` 选项，此时若震源深度小于 1 km （1000 m），
```

File: source/taup/taup_setsac.rst [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+   该命令会给出警告。
+
+   需要注意，一定不要按 :kbd:`Ctrl+C` 终止执行命令。因为 ``taup_setsac``
+   需要将 SAC 文件读入内存，进行修改，再将内存中的数据回磁盘覆盖原文件。
```
Comment (core-man) [inline source/taup/taup_setsac.rst:37]:
```suggestion
   需要将 SAC 文件读入内存，进行修改，再将内存中的数据写回磁盘覆盖原文件。
```

**Review by core-man — APPROVED** [review]

**Comment by seisman** [comment]
I just quickly read the taup_setsac manual. It seems more details need to be added. Will do it later, so WIP.

**Comment by seisman** [comment]
@core-man I just made some changes. Please give it a review.

**Comment by seisman** [comment]
😃 I enabled auto-merge in this PR, but forgot that you already approved previously. So this PR is automatically approved without your further review.

Please give it another review. We can fix it in another PR if there are any issues.

**Pull request merged**